geomstats/special_orthogonal_group.py

```diff
@@ -480,165 +526,243 @@ def matrix_from_quaternion(self, quaternion):
-        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+        if point_representation is None:
+            point_representation = self.point_representation
 
-        point = super(SpecialOrthogonalGroup, self).group_exp(
-                                     tangent_vec=tangent_vec,
-                                     base_point=base_point)
-        point = self.regularize(point)
+        base_point = self.regularize(
+            base_point, point_representation=point_representation)
+        if point_representation == 'vector':
+            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+        if point_representation == 'matrix':
+            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
+
+        if point_representation == 'vector':
+            point = super(SpecialOrthogonalGroup, self).group_exp(
+                                         tangent_vec=tangent_vec,
+                                         base_point=base_point)
+        if point_representation == 'matrix':
+            raise NotImplementedError()
+
+        point = self.regularize(
+            point, point_representation=point_representation)
         return point
 
-    def group_log(self, point, base_point=None):
+    def group_log(self, point, base_point=None, point_representation=None):
         """
         Compute the group logarithm of point point.
         """
-        point = self.regularize(point)
-        base_point = self.regularize(base_point)
+        if point_representation is None:
+            point_representation = self.point_representation
+
+        point = self.regularize(
+            point, point_representation=point_representation)
+        base_point = self.regularize(
+            base_point, point_representation=point_representation)
+
+        if point_representation == 'vector':
+            tangent_vec = super(SpecialOrthogonalGroup, self).group_log(
+                                        point=point,
+                                        base_point=base_point)
+            assert tangent_vec.ndim == 2
+
+        if point_representation == 'matrix':
+            raise NotImplementedError()
 
-        tangent_vec = super(SpecialOrthogonalGroup, self).group_log(
-                                    point=point,
-                                    base_point=base_point)
-        assert tangent_vec.ndim == 2
         return tangent_vec
 
-    def group_exponential_barycenter(self, points, weights=None):
+    def group_exponential_barycenter(
+            self, points, weights=None, point_representation=None):
         """
         Compute the group exponential barycenter in SO(n), which is the
         Frechet mean of the canonical bi-invariant metric on SO(n).
         """
-        n_points = points.shape[0]
-        assert n_points > 0
+        if point_representation is None:
+            point_representation = self.point_representation
+
+        if point_representation == 'vector':
```

Comment:
elif?

Reply:
Not here, but at a lot of other places: yes.